File: chessgame-api/src/main/java/git/dimitrikvirik/chessgameapi/model/redis/Game.java
package git.dimitrikvirik.chessgameapi.model.redis;

import git.dimitrikvirik.chessgameapi.model.game.GameMessage;
import lombok.Data;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.index.Indexed;

import java.util.ArrayList;
import java.util.List;


@RedisHash("Game")
@Data
public class Game {

    @Indexed
    private String id;
    private Integer stepNumber;
    private ChessPlayer whitePlayer;
    private ChessPlayer blackPlayer;
    private List<GameMessage> messages = new ArrayList<>();

}

File: chessgame-api/src/main/kotlin/git/dimitrikvirik/chessgameapi/model/redis/Game.kt
package git.dimitrikvirik.chessgameapi.model.redis

import git.dimitrikvirik.chessgameapi.model.Message
import org.springframework.data.redis.core.RedisHash

@RedisHash("Game")
data class Game(
    val id: String?,
    val stepNumber: Int = 0,
    val messages: List<Message> = emptyList()
) {
}